### PR TITLE
Fixes duplicated SharedProcessor#shouldResync calls

### PR DESCRIPTION
**Pull request by yue9944882**
Fixes: https://github.com/kubernetes-client/java/issues/1721
@ydmao

### CODE REVIEW

**Comment by ydmao** [comment]
this is great - thank you! When will we have the next release with this fix in?

File: util/src/main/java/io/kubernetes/client/informer/ResyncRunnable.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public ResyncRunnable(DeltaFIFO store, Supplier<Boolean> shouldResyncFunc) {
   public void run() {
     log.debug("ResyncRunnable#resync ticker tick");
 
-    if (shouldResyncFunc == null || (shouldResyncFunc.get() != null && shouldResyncFunc.get())) {
+    if (shouldResyncFunc == null || Boolean.TRUE.equals(shouldResyncFunc.get())) {
```
Comment (codeyq) [inline util/src/main/java/io/kubernetes/client/informer/ResyncRunnable.java:39]: Mind also adding a regression test in `util/src/test/java/io/kubernetes/client/informer/ResyncRunnableTest.java`?

**Review by codeyq — COMMENTED** [review]
Thank you for the quick turnaround. Could this PR be back-ported to the 12.0.0 release branch?

**Comment by brendandburns** [comment]
looks good, but adding a unit test makes sense.

File: util/src/main/java/io/kubernetes/client/informer/ResyncRunnable.java (hunk 1 above)
Comment (yue9944882) [reply util/src/main/java/io/kubernetes/client/informer/ResyncRunnable.java:39]: SGTM

**Pull request merged**